crates/stages/src/stages/bodies.rs

```diff
@@ -133,6 +167,23 @@ impl<DB: Database, D: BodyDownloader> Stage<DB> for BodyStage<D> {
                     BlockResponse::Empty(_) => 0,
                 },
             };
+
+            // Increment block on snapshot header.
+            if block_number > 0 {
+                let appended_block_number =
+                    snapshotter.increment_block(SnapshotSegment::Transactions)?;
+
+                if appended_block_number != block_number {
+                    // This scenario indicates a critical error in the logic of adding new
+                    // items. It should be treated as an `expect()` failure.
+                    return Err(StageError::InconsistentBlockNumber {
+                        segment: SnapshotSegment::Transactions,
+                        database: block_number,
+                        static_file: appended_block_number,
+                    });
+                }
```

Comment:
when does this happen? this implies to me that it might be better that `increment_block` returns a `Result`?